products/products.service.ts

```diff
@@ -0,0 +1,56 @@
+import { Injectable ,NotFoundException} from "@nestjs/common";
+import { Product } from "./products.model";
+@Injectable()
+export class productService{
+   private products:Product[]=[];
+
+   insertProduct(title:string,description:string,price:number){
+      const prodId = Math.random().toString()
```

Comment:
why you are using Math. random().it will generate the random id, but the id should be in sequence and autoincrement. 
try another way to do it.